site/src/pages/ManagementSettingsPage/CustomRolesPage/CreateEditRolePageView.tsx

```diff
@@ -0,0 +1,312 @@
+              label="Name"
+            />
+            <TextField
+              {...getFieldHelpers("display_name", {
+                helperText: "Optional: keep empty to default to the name.",
+              })}
+              fullWidth
+              label="Display Name"
+            />
+            <ActionCheckboxes
+              permissions={role?.organization_permissions || []}
+              form={form}
+              allResources={allResources}
+            />
+          </FormFields>
+        </FormSection>
+        {canAssignOrgRole && (
+          <FormFooter
+            onCancel={onCancel}
+            isLoading={isLoading}
+            submitLabel={role !== undefined ? "Save" : "Create Role"}
+          />
+        )}
+      </VerticalForm>
+    </>
+  );
+};
+
+interface ActionCheckboxesProps {
+  permissions: readonly Permission[] | undefined;
+  form: ReturnType<typeof useFormik<Role>> & { values: Role };
+  allResources: boolean;
+}
+
+const ResourceActionComparator = (
+  p: Permission,
+  resource: string,
+  action: string,
+) =>
+  p.resource_type === resource &&
+  (p.action.toString() === "*" || p.action === action);
+
+const DEFAULT_RESOURCES = [
+  "audit_log",
+  "group",
+  "template",
+  "organization_member",
+  "provisioner_daemon",
+  "workspace",
+];
+
+const resources = new Set(DEFAULT_RESOURCES);
+
+const filteredRBACResourceActions = Object.fromEntries(
+  Object.entries(RBACResourceActions).filter(([resource]) =>
+    resources.has(resource),
+  ),
+);
+
+const ActionCheckboxes: FC<ActionCheckboxesProps> = ({
+  permissions,
+  form,
+  allResources,
+}) => {
+  const [checkedActions, setCheckActions] = useState(permissions);
+  const [showAllResources, setShowAllResources] = useState(allResources);
+
+  const handleActionCheckChange = async (
+    e: ChangeEvent<HTMLInputElement>,
+    form: ReturnType<typeof useFormik<Role>> & { values: Role },
+  ) => {
+    const { name, checked } = e.currentTarget;
+    const [resource_type, action] = name.split(":");
+
+    const newPermissions = checked
+      ? [
+          ...(checkedActions ?? []),
+          {
+            negate: false,
+            resource_type: resource_type as RBACResource,
+            action: action as RBACAction,
+          },
+        ]
+      : checkedActions?.filter(
+          (p) => p.resource_type !== resource_type || p.action !== action,
+        );
+
+    setCheckActions(newPermissions);
+    await form.setFieldValue("organization_permissions", newPermissions);
+  };
+
+  const resourceActions = showAllResources
+    ? RBACResourceActions
+    : filteredRBACResourceActions;
+
+  return (
+    <>
+      <TableContainer>
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableCell>Permission</TableCell>
+              <TableCell
+                align="right"
+                sx={{ paddingTop: 0.4, paddingBottom: 0.4 }}
+              >
+                <FormControlLabel
```

Comment:
I love the addition of this! could we add something similar to the bottom of the table too, just to make sure people can find it? a small extra row that just says "show all permissions" like this does

Reply:
I added the show all permissions checkbox in a table footer row